apps/music/elements/music-search-box.js

```diff
@@ -88,50 +48,37 @@ proto.createdCallback = function() {
   var $id = shadowRoot.getElementById.bind(shadowRoot);
 
   this.els = {
-    form:  $id('form'),
+    inner: $id('inner'),
     input: $id('input'),
-    clear: $id('clear'),
     close: $id('close')
   };
 
+  this.isOpen = false;
+
   this.els.results = document.querySelector(this.getAttribute('results'));
 
   var onSearch = debounce(() => {
-    this.dispatchEvent(new CustomEvent('search', {
-      detail: this.els.input.value
-    }));
+    this.emit('search', this.els.input.value);
   }, 500);
 
-  this.els.form.addEventListener('click', (evt) => {
-    var button = evt.target.closest('button');
-    switch (button && button.id) {
-      case 'clear':
-        this.clear();
-        onSearch();
-        break;
-      case 'close':
-        this.close();
-        break;
-    }
-  });
-
   this.els.input.addEventListener('focus', () => this.open());
-
+  this.els.close.addEventListener('click', () => this.close());
   this.els.input.addEventListener('input', onSearch);
-  this.els.input.addEventListener('keypress', onSearch);
+  this.addEventListener('touchmove', e => e.preventDefault());
 
-  this.onDOMRetranslated = () => {
-    document.l10n.translateFragment(shadowRoot);
+  this.localize = () => {
+    document.l10n.formatValue('search')
+      .then(result => this.els.input.placeholder = result);
```

Comment:
I'm ok reconsidering it. I'm a bit worried about whitelist per element since it increases the noise in MutationObserver, but except of that, sounds good.